src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
```

Comment:
You can use destructuring 'for (const { type, extraData, keysToRemove } of actions)'. After that you dont need to repeat 'action.' in action.extraData, action.type, action.keysToRemove